### PR TITLE
Spring Cleaning

### PR DESCRIPTION
**Pull request by TWiStErRob**
## Description
Mostly javac/LINT/gradle/IDEA warnings, typos, and consistency fixes.

Detailed Global changes:
 * Fix some JavaDoc formatting issues (e.g. broken links)
 * Fix some JavaDoc build issues (inner class ctor links were broken)
 * Fix mis-typed words that caused grammatical issues
 * Fix Gradle deprecation warning during build
 * Make samples' build.gradle files more similar
 * Remove redundant <uses-sdk> from manifest

Mostly test related classes:
 * Add some generic arguments where missing
 * Add @TargetApi annotations to test code to suppress LINTs
 * Reduce the scope of suppressions or remove them completely
 * Use some specialized mocking/matching methods to reduce uncheckedness

GIF related classes:
 * Reduce number of magic constants, name them
 * Add some documentation from the standard for the constants
 * Remove some meaningless comments
 * Convert most inline comments to JavaDocs
 * Add annotations for ints designating color/status/etc (int is too overloaded)

There's an outstanding change in sjudd/disklrucache#4 to be fully warning free during build.

## Motivation and Context
I have a little OCD when it comes to warnings... they, must, go!
Everything should be trivial changes only, and most of them are in tests.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -65,9 +65,8 @@ subprojects { project ->
 
     gradle.projectsEvaluated {
         tasks.withType(JavaCompile) {
-            if (!name.contains('Test')) {
-                options.compilerArgs << '-Xlint:unchecked' << '-Xlint:deprecation'
-            }
+            // don't ignore because the test code is now warning-free: if (!name.contains('Test'))
```
Comment (sjudd) [inline build.gradle]: You can remove the comment too, no need to reference history

**Comment by sjudd** [comment]
Tiny nit, otherwise LGTM

**Comment by TWiStErRob** [comment]
Fix, and also included and update to sjudd/DiskLruCache#4's merge commit.

**Comment by sjudd** [comment]
Thanks!

**Pull request merged**